src/index.js

```diff
@@ -91,113 +78,126 @@ const cache = new InMemoryCache({
   },
 });
 
-const client = new ApolloClient({
-  name: '@toiletmap/ui',
-  version: version,
-  link: ApolloLink.from([errorLink, authLink, httpLink]),
-  connectToDevTools: true,
-  cache,
-  ...localSchema,
-});
-
-// Set the initial cache state
-function writeInitialState() {
-  let isAuthed = auth.isAuthenticated();
-  cache.writeQuery({
-    query: gql`
-      query {
-        mapZoom
-        mapRadius
-        mapCenter {
-          lat
-          lng
-        }
-        userData {
-          loggedIn
-          name
-        }
+ReactDOM.render(
+  <AuthProvider>
```

Comment:
I'm confused about doing the apolloClient setup inside the dom renderer.

I'm struggling to put my finger on quite why but it feels... odd somehow?

Am I worrying over nothing?

Reply:
I know what you mean. It feels weird to me, so much so I messaged Steve about the render prop approach. Do you know of another solution?

I guess we could hide the setup with something like this:

`index.js`:
```
<AuthProvider>
  <MyApolloProvider>
```

`MyApolloProvider.js`:
```
const auth = useAuth();
const client = new ApolloClient({ .. });

<ApolloClient client={client}>
```

Reply:
Perhaps like this https://github.com/neontribe/gbptm/compare/auth...auth-app?expand=1

Reply:
Well that's certainly less disturbing!